src/lib/paymentOptions/NimiqPaymentOptions.ts

```diff
@@ -81,24 +90,29 @@ export class ParsedNimiqDirectPaymentOptions extends ParsedPaymentOptions<Curren
             fee = parseInt(toNonScientificNumberString(options.protocolSpecific.fee), 10);
         }
 
-        const requiresExtendedTransaction = extraData.byteLength > 0
-            || (recipientType !== undefined && recipientType !== Nimiq.Account.Type.BASIC)
-            || (flags !== undefined && flags !== Nimiq.Transaction.Flag.NONE);
-        // Note that the transaction size can be bigger than this, for example if the sender type the user wants to use
-        // requires an extended transaction or if an extended transaction includes a multi signature proof. The size
-        // is therefore just an estimate. In the majority of cases the estimate will be accurate though and a fee that
-        // is slightly off will generally not be a problem.
-        const estimatedTransactionSize = requiresExtendedTransaction ? 166 + extraData.byteLength : 138;
-
-        if (fee === undefined) {
-            if (feePerByte === undefined) {
-                feePerByte = 0;
-                fee = 0;
+        if (enforceDefaultValues) {
+            const requiresExtendedTransaction = extraData && extraData.byteLength > 0
+                || (recipientType !== undefined && recipientType !== Nimiq.Account.Type.BASIC)
+                || (flags !== undefined && flags !== Nimiq.Transaction.Flag.NONE);
+            // Note that the transaction size can be bigger than this, for example if the sender type the user wants
+            // to use requires an extended transaction or if an extended transaction includes a multi signature proof.
+            // The size is therefore just an estimate. In the majority of cases the estimate will be accurate though
+            // and a fee that is slightly off will generally not be a problem.
+            const estimatedTransactionSize = requiresExtendedTransaction
+                ? 166 + (extraData ? extraData.byteLength : 0)
+                : 138;
+
+            if (fee === undefined) {
+                // xxx
+                if (feePerByte === undefined) {
+                    feePerByte = 0;
+                    fee = 0;
+                } else {
+                    fee = Math.ceil(feePerByte * estimatedTransactionSize);
+                }
             } else {
-                fee = Math.ceil(feePerByte * estimatedTransactionSize);
+                feePerByte = fee / estimatedTransactionSize;
             }
```

Comment:
I think this will now not update the `fee` correctly if only `feePerByte` is provided in an update (and vice versa).